ext/LegendGeSimExt.jl

```diff
@@ -0,0 +1,567 @@
+        # invert the order of the values because with siggen z=0 is tail (dirtiest part)
+        # i.e. z = L in our system (Mirion measurements)
+        imp[Npoints-i+1] = -(a + b * zpoints[i] + c * exp((zpoints[i] - L)/tau)) / 10.
+    end
+
+    crystal_name = first(meta.name, length(meta.name)-1)
+    imp_filename = joinpath("cache", crystal_name*".dat")
+	open(imp_filename, "w") do io
+		write(io, imp)
+	end    
+    @info "Impurity profile for siggen saved in $(imp_filename)"
+
+    # return name of impurity profile file and offset
+    # offset from seed end
+    det_z0 = crystal_metadata.slices[Symbol(meta.production.slice)].detector_offset_in_mm    
+
+    # siggen offset is from tail i.e. at L offset = 0
+    imp_filename, L - det_z0
+end
+
+"""
+Simulation method: siggen
+"""
+
+
+
+function LegendGeSim.simulate_waveforms(stp_events::Table, detector::SigGenSetup, ::SiggenSimulator)
+    T = Float32 # This should be somehow defined and be passed properly
+    @info("~.~.~ Siggen")
+
+    nevents = length(stp_events)
+    wf_array = Array{RDWaveform}(undef, nevents)
+    for i in 1:nevents
+        # println("$i/$nevents")
+        if(i % 100  == 0) println("$i/$nevents") end
+        signal::Vector{Float32} = simulate_signal(stp_events[i].pos, stp_events[i].edep, detector)
+        time = range(T(0)u"ns", step = detector.step_time_out*u"ns", length = length(signal))
+
+        # push!(waveforms, signal)
+        wf_array[i] = RDWaveform(time, signal)
+    end
+
+    ## Oliver said this should be the more efficient way of doing it
+    ## as opposed to an array of separate RDWaveform instances
+    # Δt = setup.step_time_out*u"ns"
+    # t_end = length(waveforms[1]) * Δt
+    # times = fill(0u"ns" : Δt : t_end, length(waveforms))
+    # ArrayOfRDWaveforms((times, VectorOfSimilarVectors(waveforms)))
+
+    waveforms = ArrayOfRDWaveforms(wf_array)
+    # why am I doing this?
+    waveforms = ArrayOfRDWaveforms((waveforms.time, VectorOfSimilarVectors(waveforms.signal)))    
+
+    pss_table = Table(
+        channel = [1 for idx in 1:length(waveforms)], # lists of ADCs that triggered, 1 for HADES all the time
+        ievt = stp_events.evtno,
+        waveform = waveforms
+    )
+
+    # using SSD we get mc truth from SSD output
+    # with siggen, let's just return the input
+    # maybe mc truth will not be propagated to (sim)raw anyway
+    pss_truth = Table(
+        detno = stp_events.detno,
+        edep = stp_events.edep,
+        ievt = stp_events.evtno,
+        pos = stp_events.pos,
+        thit = stp_events.thit        
+    )
+
+    pss_table, pss_truth
+end
+
+
+
+"""
+    simulate_wf(pos, edep, siggen_setup)
+
+AbstractVector, AbstractVector, SigGenSetup -> Vector{Float32}
+
+Simulate a signal from energy depositions <edep> with corresponding positions <pos>
+    based on a given <siggen_setup>    
+"""
+
+function simulate_signal(pos::AbstractVector, edep::AbstractVector, siggen_setup::SigGenSetup)
```

Comment:
The docstring does not match the function.

Also: Not sure whether this should be `LegendGeSim.simulate_signal` instead of `simulate_signal`.